numba/datamodel/models.py

```diff
@@ -491,6 +490,17 @@ def from_data(self, builder, value):
         return NotImplemented
 
 
+@register_default(types.NestedArray)
+class NestedArrayModel(ArrayModel):
+    def __init__(self, dmm, fe_type):
+        self._be_type = dmm.lookup(fe_type.dtype).get_data_type()
+        super(NestedArrayModel, self).__init__(dmm, fe_type)
+
+    def get_data_type(self):
```

Comment:
In the CUDA target, `BaseContext.sentry_record_type` (numba/targets/base.py:866) is called to ensure that the offset of all Record members is a multiple of the size of their data type. If I understand correctly, the data type of an array type is the struct holding information about the array including the pointer, shape, strides, etc. So here `get_data_type` returns the element's data type so that the check checks alignment w.r.t. one element, and not the structure describing the `NestedArray` data.